services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/attachment/common/FileScanService.kt

```diff
@@ -30,14 +32,26 @@ class FileScanService(
             logger.info { "Initialized file scan client with base-url: $clamAvUrl" }
         }
 
-    fun scanFiles(files: List<Pair<String, ByteArray>>): List<FileResult> {
+    fun validate(file: MultipartFile): MultipartFile {
+        val scanInput = validScanInput(file)
```

Comment:
I don't think the business rules we have for what kind of attachments we allow should be a prerequisite for doing a virus scan. It feels like we're coupling two things that shouldn't have much to do with each other.

Reply:
My idea was to avoid duplication across application / hanke implementations. But yes, it might be better to keep these separate. I'll see what I can do.

Reply:
Separated these two.